File: day1.js
import {readFileSync, promises as fsPromises} from 'fs'

function syncReadFile(filename) {
    const content = readFileSync(filename, 'utf8')
    const elfArr = content.split(/\r\n\r\n/)
    const caloriesArr = elfArr.map(x => x.split(/\r\n/))
    return caloriesArr.map(x => x.map(y=>Number(y)))    // array of elfs each contains an array of calories (interger)
}

const input = syncReadFile('./input.txt')
const sortedCalories = input.map(x=>x.reduce((acc, curr)=>acc+curr)).sort((a,b)=>b-a)
console.log('Top calories Elf: ' + sortedCalories[0])
let top3 = 0
for (let i=0; i<3; i++) {
    top3 += sortedCalories[i]
}
console.log('Top 3 sum of calories: ' + top3)

File: day1.py
import functools

elfList = open('./input.txt').read().split('\n\n')
elfsCalories = []
elfs2dArray = []
for elf in elfList:
    arr = elf.split('\n')
    elfs2dArray.append(arr)
    elfsCalories.append(
    int(functools.reduce(lambda a, b: int(a) + int(b), arr))
    )
max = functools.reduce(lambda a, b: int(a) if int(a) > int(b) else int(b), elfsCalories)
print('Elf carrying the most Calories:',max)
max_index = elfsCalories.index(max)
print('Elf index:',max_index)
print('Elf calories:',elfs2dArray[max_index])
caloriresRank = sorted(elfsCalories,reverse=True)
top3 = 0
for i in range(0,3):
    top3 += int(caloriresRank[i])
print('Sum of top 3 (calories):',top3)